clients/src/main/java/org/apache/kafka/common/record/EosLogEntry.java

```diff
@@ -57,7 +57,8 @@
     static final int RECORDS_OFFSET = BASE_SEQUENCE_OFFSET + BASE_SEQUENCE_LENGTH;
     public static final int LOG_ENTRY_OVERHEAD = RECORDS_OFFSET;
 
-    private static final int COMPRESSION_CODEC_MASK = 0x07;
+    private static final byte COMPRESSION_CODEC_MASK = 0x07;
```

Comment:
Ack. We did it for EosLogRecord, but not here yet. I'll go ahead and add it to this patch.